shell_automaton/src/mempool/mempool_state.rs

```diff
@@ -69,6 +69,71 @@ impl MempoolState {
             .get(&peer)
             .map_or(false, |p| p.seen_operations.contains(op_hash))
     }
+
+    /// Get next operation with highest priority for prevalidation.
+    pub fn next_for_prevalidation(&self) -> Option<(&OperationHash, &Operation)> {
+        self.injecting_rpc_ids
+            .iter()
+            .find_map(|(hash, _)| {
+                self.pending_operations
+                    .get(hash)
+                    .map(|content| (hash, content))
+            })
+            .or_else(|| self.pending_operations.next_for_prevalidation())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct MempoolPendingOperations {
+    ops: HashMap<OperationHash, Operation>,
+    consensus_ops: VecDeque<OperationHash>,
```

Comment:
👌  for our current needs, this is perfect. Just one suggestion, wouldn't it be better to use a BTreeSet instead of a VecDeque? I see bellow that we need to iterate it comparing hashes to remove the entry when the operation is removed, which doesn't feel ideal.

Reply:
@tizoc  I used `VecDeque` to guarantee FIFO order. I was thinking on using `BTreeSet`, but changed my mind because we don't really need that.

Only times when we need to remove an element from that:
- When we have validated an operation. It will always be first operation in that queue (removing of which is `O(1)`), unless it was injected + non-consensus operation that was validated. In that case yeah, we will iterate on whole list for no reason.
- When new block is applied. This happens every 15s and only once so I wouldn't worry about that being slow.

Reply:
Ok, sounds good, only thing I would add then is a comment mentioning this in `remove`, for our future selves.